src/components/DeviceForm/index.js

```diff
@@ -31,28 +32,32 @@ class DeviceForm extends Component {
     render() {
         const { form } = this.state
         return (
-            <FormWrapper onSubmit={this.handleSubmit}>
-                <FormInput
-                    desc="Name"
-                    name="name"
-                    value={form.name || ''}
-                    onChange={this.handleChange('name')}
-                />
-                <FormInput
-                    desc="IP"
-                    isFunny
-                    value={form.ip || ''}
-                    name="ip"
-                    onChange={this.handleChange('ip')}
-                />
-                <FormInput
-                    desc="Description"
-                    name="description"
-                    value={form.description || ''}
-                    onChange={this.handleChange('description')}
-                />
-                <input type="submit" value="Submit" />
-            </FormWrapper>
+            <DevicesContext.Consumer>
+                {value => (
```

Comment:
If you only use onSubmit from here, then it's better that you destructure it directly from value